neevaco/neeva-ios
# Remove Indexed Browsing Activities When Deleting Browsing History

Add calls to delete indexed `NSUserActivity`s from Spotlight when browsing history has been cleared.
This applies to both clearing from `HistoryPanel` and `DataManagementView`(Settings).

HistoryPanel can clear history by recent time period. Since we don't keep identifiers to the activity items, we delete all browsing items regardless of the time period.

Also added an abstract interface from UserActivityHandler to remove items.

## Checklists

### How was this tested?

### Manual test cases
Opened some webpages. Check they are in spotlight. 
Delete history in settings. Check they are removed from spotlight (a new search is needed since spotlight keeps results between launches)
Repeat and clear in history panel.

### Screenshots and screen recordings

## Issues addressed
